### DISCUSSION TITLE
0% code coverage when code should be covered

### DISCUSSION THREAD

**Discussion by RomainBitard** [nunit/nunit#4121] [Q&A] [answered]
Hello everyone,

my code coverage is at 0 and I have asserts and the code is executed (if I launch my tests in debug I can see the test go through the production code).
I have no error message do you know where I can start to debug this problem ? 
I'm using Rider and Unity3D but my unit tests are decoupled from unity using assembly definitions and no references to unity 

Cordially,
Romain

**Comment by mikkelbu** [comment, marked as answer]
Probably the best place to start is to examine the tool you use for code coverage. NUnit don't provide any framework for code coverage, so the 0% is coming from somewhere else. Since you are using Rider I guess you are using dotCover, so I would search for why dotCover don't provide any coverage information.

**Comment by RomainBitard** [comment]
Thanks for the answer ! I'm not well versed in .net tooling, I'll check
that, effectively I saw dotcover on rider
